### DISCUSSION TITLE
Supabase Subscription Complexity

### DISCUSSION THREAD

**Discussion by sswatson** [supabase/supabase#17701] [Questions]
With Firebase, I can subscribe to a query in the client and count on the client to reflect the result of that query at all times (unless the user loses internet or something).

With Supabase, it feels like I have a lot more complexity to manage:

1. Separate handling for the initial query and processing the change feed.
2. Checking for changes that might have been missed in intervals while the websocket was reconnecting.
3. Reloading the page data on events like device sleep/wake.

I find that this additional complexity really adds up, and I feel like I'm solving very general problems without a lot of confidence that I'm doing everything the right way.

I'm curious if other folks have encountered similar challenges, and whether perhaps I'm approaching the way I use Supabase incorrectly.

**Comment by GaryAustin1** [comment]
Those sound like the things I've had to deal with.  
https://github.com/GaryAustin1/Realtime2
Another user just added this, but it does not deal with the restarts yet, but looks reasonable.
https://github.com/openartmarket/supabase-live-table/blob/main/src/index.ts
